src/test/lit-element_test.ts

```diff
@@ -1243,26 +1310,238 @@ suite('LitElement', () => {
+    assert.equal(sub.shadowRoot!.textContent, '10');
+  });
+
+  test('user accessors only using noAccessor', async () => {
+    class E extends LitElement {
+      _updateCount = 0;
+      _foo?: String;
+      _bar?: String;
+      static get properties() {
+        return {
+          foo : {type : String, reflect : true, noAccessor : true},
+          bar : {type : String, reflect : true, noAccessor : true}
+        };
+      }
+      constructor() {
+        super();
+        this.foo = 'defaultFoo';
+        this.bar = 'defaultBar';
+      }
+      set foo(value: string) {
+        const old = this._foo;
+        this._foo = value;
+        this.requestUpdate('foo', old);
+      }
+      get foo() { return this._foo as string; }
+      set bar(value: string) {
+        const old = this._bar;
+        this._bar = value;
+        this.requestUpdate('bar', old);
+      }
+      get bar() { return this._bar as string; }
+      update(changedProperties: PropertyValues) {
+        this._updateCount++;
+        super.update(changedProperties);
+      }
+      render() { return html`${this.foo}-${this.bar}`; }
+    }
+    customElements.define(generateElementName(), E);
+
+    const el = new E();
+    el.foo = 'foo1';
+    document.body.appendChild(el);
+    await el.updateComplete;
+    assert.equal(el.foo, 'foo1');
+    assert.equal(el.bar, 'defaultBar');
+    assert.equal(el.getAttribute('foo'), 'foo1');
+    assert.equal(el.getAttribute('bar'), 'defaultBar');
+    assert.equal(el.shadowRoot!.textContent, 'foo1-defaultBar');
+    assert.equal(el._updateCount, 1);
+
+    el.foo = 'foo2';
+    el.bar = 'bar';
+    await el.updateComplete;
+    assert.equal(el.foo, 'foo2');
+    assert.equal(el.bar, 'bar');
+    assert.equal(el.getAttribute('foo'), 'foo2');
+    assert.equal(el.getAttribute('bar'), 'bar');
+    assert.equal(el.shadowRoot!.textContent, 'foo2-bar');
+    assert.equal(el._updateCount, 2);
+
+    el.foo = 'foo3';
+    await el.updateComplete;
+    assert.equal(el.foo, 'foo3');
+    assert.equal(el.getAttribute('foo'), 'foo3');
+    assert.equal(el.getAttribute('bar'), 'bar');
+    assert.equal(el.shadowRoot!.textContent, 'foo3-bar');
+    assert.equal(el._updateCount, 3);
+  });
+
+  test('attributeChangedCallback based rendering', async () => {
+    class E extends LitElement {
+      _updateCount = 0;
+      static get properties() {
+        return {
+          foo : {type : String, noAccessor : true},
+          bar : {type : String, noAccessor : true}
+        };
+      }
+      set foo(value: string|null) { this.setAttribute('foo', value as string); }
+      get foo() { return this.getAttribute('foo') || 'defaultFoo'; }
+      set bar(value: string|null) { this.setAttribute('bar', value as string); }
+      get bar() { return this.getAttribute('bar') || 'defaultBar'; }
+      attributeChangedCallback(name: string, old: null|string,
+                               _value: null|string) {
+        this.requestUpdate(name, old);
```

Comment:
Let's call `super.attributeChangedCallback` here because otherwise an extension would not be compatible with `LitElement`'s reflect system and it might be confusing if someone looked at this as an example.

Reply:
Done